### PR TITLE
fix: eigen context

### PR DESCRIPTION
**Pull request by ds300**
Web views in eigen have been slightly broken for some time. The root cause was that the `isEigen` system context value was not managed correctly. This PR fixes that while making sure that the main nav bars are hidden in eigen.

image
image

### CODE REVIEW

File: src/v2/Components/NavBar/NavBar.tsx [hunk 1]
```diff
@@ -47,8 +47,11 @@ export const NavBar: React.FC = track(
     dispatch: data => Events.postEvent(data),
   }
 )(() => {
+  const { mediator, user, isEigen } = useSystemContext()
+  if (isEigen) {
+    return null
+  }
```
Comment (damassi) [inline src/v2/Components/NavBar/NavBar.tsx:53]: 👍

**Comment by damassi** [comment]
👍 👍

**Comment by MounirDhahri** [comment]
🎉

File: src/v2/Components/NavBar/MinimalNavBar.tsx [hunk 2]
```diff
@@ -19,19 +21,21 @@ export const MinimalNavBar: React.FC<MinimalNavBarProps> = props => {
...
+              // TODO: figure out a minimal example of the underlying cause of this error
+              // and submit an issue to TS 😓
+              // @ts-ignore
+              data-test="logoLink"
+            >
+              <ArtsyLogoBlackIcon />
+            </RouterLink>
+          </Box>
+        </AppContainer>
+      )}
```
Comment (pvinis) [inline src/v2/Components/NavBar/MinimalNavBar.tsx:38]: I was too slow for this, but I would prefer something like:
```
      {isEigen ? (
        null
      ) : (
        <AppContainer>
          <Box height={70} px={[2, 4]}>
            <RouterLink
              to={props.to}
              // TODO: figure out a minimal example of the underlying cause of this error
              // and submit an issue to TS 😓
              // @ts-ignore
              data-test="logoLink"
            >
              <ArtsyLogoBlackIcon />
            </RouterLink>
          </Box>
        </AppContainer>
      )}
```

Comment (damassi) [reply src/v2/Components/NavBar/MinimalNavBar.tsx:38]: or maybe even better:

```ts
 if (isEigen) {
   return null
 }
 
 return ... 
``

Comment (pvinis) [reply src/v2/Components/NavBar/MinimalNavBar.tsx:38]: Sure, of course. It's just inline and that's why I would go for the `check ? null : rest`, but either way. Anyway, cool PR 😃!

**Pull request merged**